bin/kano_tzupdate

```diff
@@ -0,0 +1,156 @@
+import json
+import os
+import sys
+
+
+# Accomodate a library to speak to the web services
+try:
+    from urllib.request import urlopen
+    from urllib.parse import urlencode
+except ImportError:  # Python 2 fallback
+    from urllib import urlopen, urlencode
+
+
+# Authentication information required by the web services
+AUTH = {
+    "geonames": {
+        "username": "kanosystem",
+    },
+}
+
+PATH = {
+    "zoneinfo": "/usr/share/zoneinfo",
+    "localtime": "/etc/localtime",
+}
+
+
+def get_geolocation(ip=None):
+    '''
+    This is a simplified version to obtain your current location
+    by using the free web service ipinfo.io API
+    '''
+    lat=lng=None
+
+    if ip:
+        url='http://ipinfo.io/{}/geo'.format(ip)
+    else:
+        url="http://ipinfo.io/geo"
+
+    try:
+        result = urlopen(url)
+        res_data = json.loads(result.read())
+        lat,lng=res_data['loc'].split(',')
+        return lat,lng
+    except:
+        raise
+
+
+def timezone_from_location(coords):
+    """
+    Return the timezone for a set of coordinates using the GeoNames API.
+
+    :param coords: latitude and longitude
+    :returns: timezone, or None if none is available
+    """
+
+    url = "http://api.geonames.org/timezoneJSON?"
+
+    latitude, longitude = coords
+    url_params = {
+        "username": AUTH["geonames"]["username"],
+        "lat": latitude,
+        "lng": longitude,
+    }
+
+    res_handle = urlopen(url + urlencode(url_params))
+
+    try:
+        res_encoding = res_handle.headers.get_content_charset()
+    except AttributeError:  # Python 2 fallback
+        res_encoding = res_handle.headers.getparam("charset")
+
+    res_data = json.loads(res_handle.read().decode(res_encoding))
+
+    timezone = res_data.get("timezoneId")
+
+    return timezone
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "-p", "--print-only",
+        action="store_true",
+        help="print the timezone, but don't update " + PATH["localtime"]
+    )
+    parser.add_argument(
+        "-a", "--ip",
+        help="use this IP instead of automatically detecting it"
+    )
+    args = parser.parse_args()
+
+    try:
+        coords = get_gelocation(ip=args.ip)
+    except LookupError as e:
+        print "IPInfoDB returned an error: " + str(e)
+        sys.exit(3)
+
+    timezone = timezone_from_location(coords)
+
+    if timezone is None:
+        print "No timezone available for this IP."
+        sys.exit(2)
+
+    if not args.print_only:
+        timezone_path = os.path.join(PATH["zoneinfo"], timezone)
+        if not os.path.isfile(timezone_path):
+            print "Unknown timezone: " + timezone_path
+            sys.exit(1)
+
+        try:
+            os.unlink(PATH["localtime"])
+        except OSError as e:
+            if e.errno != errno.ENOENT:
+                raise
+
+        os.symlink(timezone_path, PATH["localtime"])
+
+    print(timezone)
```

Comment:
This print will give us the country code which is caught by the udhcp script to set the timezone under /etc/localtime, below (which needs to change to `kano_tzupdate`:

 * https://github.com/KanoComputing/kano-toolset/blob/master/udhcpc/kano.script#L95

@alex5imon @Ealdwulf